agent/config/config_accessor.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+// agentConfigAccessor struct implements AgentConfigAccessor interface defined in ecs-agent module.
+type agentConfigAccessor struct {
+	cfg *Config
+}
+
+// NewAgentConfigAccessor creates a new agentConfigAccessor.
+func NewAgentConfigAccessor(cfg *Config) *agentConfigAccessor {
+	return &agentConfigAccessor{
+		cfg: cfg,
+	}
+}
+
+func (aca agentConfigAccessor) APIEndpoint() string {
```

Comment:
if `NewAgentConfigAccessor` is returning `*agentConfigAccessor`, all the interface methods' signature should be

```
func (aca *agentConfigAccessor) Method() {...}
```
?

Reply:
Thanks for calling out! I will update this.